src/components/PipelineRun/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts

```diff
@@ -164,78 +254,68 @@ export const appendStatus = (
+      return;
     }
     if (!taskRuns || taskRuns.length === 0) {
-      return { ...task, status: { reason: overallPipelineRunStatus } };
+      result.push({ ...task, status: { reason: overallPipelineRunStatus } });
+      return;
     }
 
-    const taskRun = taskRuns.find(
-      (tr) => tr.metadata.labels[TektonResourceLabel.pipelineTask] === task.name,
-    );
-    const taskStatus: TaskRunStatus = taskRun?.status;
-    const taskResults = isTaskV1Beta1(taskRun)
-      ? taskRun?.status?.taskResults
-      : taskRun?.status?.results;
-
-    const mTask: PipelineTaskWithStatus = {
-      ...task,
-      status: { ...taskStatus, reason: runStatus.Pending },
-    };
-
-    // append task duration
-    if (mTask.status.completionTime && mTask.status.startTime) {
-      const date =
-        new Date(mTask.status.completionTime).getTime() -
-        new Date(mTask.status.startTime).getTime();
-      mTask.status.duration = formatPrometheusDuration(date);
-    }
-    // append task status
-    if (!taskStatus) {
+    const taskRunsForTask = taskRunsByTaskName.get(task.name) || [];
+
+    // If no TaskRuns found, create a single task entry
+    if (taskRunsForTask.length === 0) {
       const isSkipped = !!pipelineRun.status.skippedTasks?.find((t) => t.name === task.name);
-      if (isSkipped) {
-        mTask.status.reason = runStatus.Skipped;
-      } else {
-        mTask.status.reason = runStatus.Idle;
-      }
-    } else if (mTask.status.conditions) {
-      mTask.status.reason = taskRunStatus(taskRun);
+      result.push({
+        ...task,
+        status: { reason: isSkipped ? runStatus.Skipped : runStatus.Idle },
+      });
+      return;
     }
 
-    // Determine any task test status
-    if (taskResults) {
-      const testOutput: TektonResultsRun = taskResults.find(
-        (result) => result.name === 'HACBS_TEST_OUTPUT' || result.name === 'TEST_OUTPUT',
-      );
-      if (testOutput) {
-        try {
-          const outputValues = JSON.parse(testOutput.value);
-          mTask.status.testFailCount = parseInt(outputValues.failures as string, 10);
-          mTask.status.testWarnCount = parseInt(outputValues.warnings as string, 10);
-        } catch (e) {
-          // ignore
-        }
-      }
-      const scanResult = taskResults?.find((result) => isCVEScanResult(result));
-
-      if (scanResult) {
-        try {
-          mTask.status.scanResults = JSON.parse(scanResult.value);
-        } catch (e) {
-          // ignore
-        }
-      }
-    }
-    // Get the steps status
-    const stepList = taskStatus?.steps || mTask?.steps || mTask?.taskSpec?.steps || [];
-    mTask.steps = stepList.map((step, i, { length }) =>
-      createStepStatus(step.name as string, mTask.status, i + 1 === length),
+    // Check if this is a matrix task (multiple TaskRuns with platform labels)
+    const platformTaskRuns = taskRunsForTask.filter(
+      (tr) => tr.metadata.labels?.['build.appstudio.redhat.com/target-platform'],
```

Comment:
nit: maybe we could add this label to a enum, the same way we have the `PipelineRunLabel` for example (`src/consts/pipelinerun.ts`). So we could have something like:

```ts
export enum TaskRun {
  TARGET_PLATFORM = 'build.appstudio.redhat.com/target-platform',
}
```

so we could simply import it and use it in all the places we're hardcoding this string here. No big deal, just a suggestion :)